src/main/java/cz/muni/fi/pv286/mmt/representation/IntParser.java

```diff
@@ -0,0 +1,140 @@
+package cz.muni.fi.pv286.mmt.representation;
+
+import cz.muni.fi.pv286.mmt.exceptions.InvalidIntCountException;
+import cz.muni.fi.pv286.mmt.exceptions.InvalidIntInputException;
+import cz.muni.fi.pv286.mmt.model.IoOption;
+import cz.muni.fi.pv286.mmt.model.Options;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+
+/**
+ * Int parser.
+ */
+public class IntParser extends RepresentationParser {
+
+    public IntParser(Options options) {
+        super(options);
+    }
+
+    private boolean isNumber(char c) {
+        return Character.digit(c, 10) != -1;
+    }
+
+    private byte[] getBytesFromInteger(int integer, Endian endian) {
+        if (endian == Endian.BIG) {
+            return new byte[] {
+                (byte) (integer >> 24),
+                (byte) (integer >> 16),
+                (byte) (integer >> 8),
+                (byte) integer
+            };
+        } else {
+            return new byte[] {
+                (byte) integer,
+                (byte) (integer >> 8),
+                (byte) (integer >> 16),
+                (byte) (integer >> 24)
+            };
+        }
+    }
+
+    private byte[] reverseByteArray(byte[] array) {
+        byte[] reversed = new byte[array.length];
+        for (int i = 0; i < array.length; i++) {
+            reversed[i] = array[array.length - i - 1];
+        }
+        return reversed;
+    }
+
+    @Override
+    public byte[] parseTo(byte[] bytes) throws IOException {
+
+        if (bytes.length > 4) {
+            throw new InvalidIntInputException();
+        }
+
+        byte[] bytesToWrite = new byte[4];
+
+        int pad = 4 - bytes.length;
+        int i;
+        for (i = 0; i < pad; i++) {
+            bytesToWrite[i] = 0;
+        }
+
+        for (byte b : bytes) {
+            bytesToWrite[i] = b;
+            i++;
+        }
+
+        Optional<IoOption> endianness = options.getOutputFromToOption();
+        if (endianness.isPresent()) {
+            if (endianness.get() == IoOption.LITTLE) {
+                bytesToWrite = reverseByteArray(bytesToWrite);
+            } // else by default is Big endian
+        }
+
+        // Java doesn't have unsigned types, so we need to apply a mask to every byte
+        long integer = ((long) (bytesToWrite[0] & 0xff) << 24)
+            + ((bytesToWrite[1] & 0xff) << 16)
+            + ((bytesToWrite[2] & 0xff) << 8)
+            + (bytesToWrite[3] & 0xff);
+        byte[] integerAsString = String.valueOf(integer).getBytes(StandardCharsets.UTF_8);
+        return integerAsString;
```

Comment:
**nitpick:** Unneccessary assignment of variable `integerAsString`

You can return the value immediately.